### PR TITLE
Enhanced RBAC Rule assignment

### PR DESCRIPTION
**Pull request by mdmunir**
| Q | A |
| --- | --- |
| Is bugfix? | no |
| New feature? | yes |
| Breaks BC? | no |
| Tests pass? | yes |
| Fixed issues | #11254 |

Directly attach rule with it class name

``` php
$role = $auth->createRole('Reader');
$role->ruleName = 'app\rbac\ActionRule';
$auth->add($role);

// using DI
Yii::$container->set('write_rule', ['class' => 'app\rbac\ActionRule', 'action' => 'write']);
$role = $auth->createRole('Writer');
$role->ruleName = 'write_rule';
$auth->add($role);
```

### CODE REVIEW

File: framework/rbac/BaseManager.php [hunk 1]
```diff
@@ -115,6 +115,11 @@ public function createPermission($name)
     public function add($object)
     {
         if ($object instanceof Item) {
+            if($object->ruleName && $this->getRule($object->ruleName) === null){
```
Comment (SilverFire) [inline framework/rbac/BaseManager.php]: Could you add spaces before and after parentheses?

`if ($object->ruleName && $this->getRule($object->ruleName) === null) {`

File: framework/rbac/BaseManager.php [hunk 2]
```diff
@@ -143,6 +148,11 @@ public function remove($object)
     public function update($name, $object)
     {
         if ($object instanceof Item) {
+            if($object->ruleName && $this->getRule($object->ruleName) === null){
```
Comment (SilverFire) [inline framework/rbac/BaseManager.php]: Same here

**Comment by SilverFire** [comment]
Great! Could you add a changelog and it's ready to merge?

**Comment by samdark** [comment]
Merged. Thanks!

**Pull request merged**